core/metacling/src/TCling.cxx

```diff
@@ -3675,7 +3675,7 @@ void TCling::SetClassInfo(TClass* cl, Bool_t reload)
 {
    // We are shutting down, there is no point in reloading, it only triggers
    // redundant deserializations.
-   if (fCxxModulesEnabled && reload && fIsShuttingDown)
+   if (fIsShuttingDown)
       return;
```

Comment:
Should we `assert(reload)`? We really shouldn't be requesting new `TClass`es when shutting down.